docs/root/configuration/http/http_filters/rate_limit_quota_filter.rst

```diff
@@ -0,0 +1,168 @@
+  domain: "acme-services"
+  matcher:
+    matcher_list:
+      matchers:
+      - predicate:
+        - single_predicate:
+            input:
+              name: request-headers
+              typed_config:
+                "@type": type.googleapis.com/envoy.type.matcher.v3.HttpRequestHeaderMatchInput
+                header_name: deployment
+            value_match:
+              exact: prod
+        on_match:
+          action:
+            name: prod-bucket
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaBucketSettings
+              bucket_id_builder:
+                bucket_id_builder:
+                  "name":
+                    string_value: "prod-rate-limit-quota"
+              reporting_interval: 60s
+              no_assignment_behavior:
+                blanket_rule: ALLOW_ALL
+      - predicate:
+        - single_predicate:
+            input:
+              name: request-headers
+              typed_config:
+                "@type": type.googleapis.com/envoy.type.matcher.v3.HttpRequestHeaderMatchInput
+                header_name: deployment
+            value_match:
+              exact: staging
+        on_match:
+          action:
+            name: staging-bucket
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaBucketSettings
+              bucket_id_builder:
+                bucket_id_builder:
+                  "name":
+                    string_value: "staging-rate-limit-quota"
+              reporting_interval: 60s
+              no_assignment_behavior:
+                blanket_rule: DENY_ALL
+    # The "catch all" bucket settings
+    on_no_match:
+      action:
+        name: default-bucket
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaBucketSettings
+          bucket_id_builder:
+            bucket_id_builder:
+              "name":
+                string_value: "default-rate-limit-quota"
+          reporting_interval: 60s
+          deny_response_settings:
+            http_status_code: 429
+          no_assignment_behavior:
+            blanket_rule: ALLOW_ALL
+          expired_assignment_behavior:
+            fallback_rate_limit:
+              requests_per_time_unit:
+                requests_per_time_unit: 1000
+                time_unit: 1s
+
+
+Rate Limit Quota Override
+-------------------------
+
+Rate limit filter :ref:`configuration <envoy_v3_api_msg_extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaFilterConfig>` can be overridden
+at the virtual host or route levels using the :ref:`RateLimitQuotaOverride <envoy_v3_api_msg_extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaOverride>`
+configuration. The more specific configuration fully overrides less specific configuration.
+
+Matcher extensions
+------------------
+
+TODO
```

Comment:
Can this TODO be written now?

Reply:
I would prefer to write this TODO later if it is ok.

We are planing to use CEL expression matcher here but this has not been implemented/tested yet. So I think it will be best to write this section at least after I have prototyped CEL expression.